sound/soc/sof/ipc.c

```diff
@@ -400,15 +400,23 @@ static void ipc_msgs_rx(struct work_struct *work)
 		if (!sdev->boot_complete) {
 			if (sof_ops(sdev)->fw_ready)
 				err = sof_ops(sdev)->fw_ready(sdev, cmd);
+			else
+				dev_warn(sdev->dev,
+					"warning: fw_ready ops not defined\n");
 			if (err < 0) {
-				dev_err(sdev->dev, "error: DSP firmware boot timeout %d\n",
+				dev_err(sdev->dev,
+					"error: fw ready IPC failed with error %d\n",
 					err);
+
+				/* fw_ready() error indicates ABI mismatch */
+				sdev->abi_mismatch = true;
+				return;
```

Comment:
the errow flow is odd, if fw_ready isn't supported then we'll go through the boot complete sequence in the else case below. Not really good.

Reply:
@plbossart fixed to report an error in case fw_ready() is not defined.